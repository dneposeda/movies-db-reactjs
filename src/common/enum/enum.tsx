export enum TypeNameBtnGroups {
    SORT = 'sort',
    SEARCH = 'search',
}

export enum TypeSearch {
    TITLE = 'title',
    GENRE = 'genres',
}

export enum SearchButtonNames {
    TITLE = 'title',
    GENRE = 'genre',
}

export enum TypeSort {
    DATE = 'release_date',
    RATING = 'vote_average',
}

export enum SortButtonNames {
    DATE = 'release date',
    RATING = 'rating'
}
