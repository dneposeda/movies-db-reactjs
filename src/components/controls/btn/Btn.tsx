import React, {FC} from 'react';
import classNames from 'classnames';

interface ButtonProps {
    title: string,
    cls?: string[],
    active?: boolean,
    onClick?: () => void,
}
type Component = FC<ButtonProps>;

const Btn: Component = ({ title , cls, active, onClick}) => (
    <button
        type="button"
        className={ classNames(cls, { active: active }) }
        onClick={ onClick }
    >
        { title }
    </button>
)

export default Btn;
