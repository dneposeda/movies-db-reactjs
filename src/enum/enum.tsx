enum TypeNameBtnGroups {
    SORT = 'sort',
    SEARCH = 'search',
}

enum TypeSearch {
    TITLE = 'title',
    GENRE = 'genre',
}

enum TypeSort {
    DATE = 'release date',
    RATING = 'rating',
}

export { TypeNameBtnGroups, TypeSearch, TypeSort };
