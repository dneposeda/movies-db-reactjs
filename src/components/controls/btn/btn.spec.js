import React from 'react';
import Btn from './Btn';
import { TypeSort } from '../../../enum/enum'
import { mount } from 'enzyme';

const mockBnt = {
    title: TypeSort.DATE,
    cls: '',
    active: true,
    handleClick: () => true,
}

describe('Btn', () => {
    it('renders correctly', () => {
        const component = mount (
            <Btn
                title={mockBnt.title}
                cls={mockBnt.cls}
                active={mockBnt.active}
                handleClick={mockBnt.handleClick}
            />
        );
        const title = component.find('[type="button"]');

        expect(component).toMatchSnapshot();
        expect(component.find('.active')).toHaveLength(1);
        expect(title.text(mockBnt.title)).toEqual(TypeSort.DATE);
    });
});
