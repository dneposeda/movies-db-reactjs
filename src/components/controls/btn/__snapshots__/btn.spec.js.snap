// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn renders correctly 1`] = `
<Btn
  active={true}
  cls=""
  handleClick={[Function]}
  title="release date"
>
  <button
    className="active"
    onClick={[Function]}
    type="button"
  >
    release date
  </button>
</Btn>
`;
